packages/frontend/src/redux/crossStateSelectors/selectCollectedAvailableForClaimData.js

```diff
@@ -0,0 +1,81 @@
+import BN from 'bn.js';
+import { isEmpty } from 'lodash';
+import { createSelector } from 'reselect';
+
+import { selectValidatorsFarmData } from '../slices/staking';
+import {
+    selectTokensFiatValueUSD,
+    selectTokenWhiteList,
+} from '../slices/tokenFiatValues';
+import { selectAllContractMetadata } from '../slices/tokens';
+
+const collectFarmingData = (...args) => {
+    try {
```

Comment:
Not really. But the catch block will return `[]` if something will go wrong. Just in case. Probably, if the app somewhere in the future will ride to TS - this can be removed.

Reply:
I see, so it can be removed. Using try/catch makes sense in situations when you are expecting that error might appear, for example, the request will throw an error, and you need to handle this error.

Here, you are basically filtering data, so if you expect that there might be a problem with particular data, you should handle it there explicitly.

Reply:
Resolve this as during my last commits catch blocks were mostly required for returning empty data.
I can't assume that something will crash here, but I think that this may save us at least from some invalid data on devnet and testned.